src/main/java/com/bio4j/model/NCBITaxonomyGraph.java

```diff
@@ -65,46 +68,54 @@
 
       Values are, among others, *species*, *genus*, or *no rank*; see the enum below.
     */
-    public final TaxonomicRank taxonomicRank = new TaxonomicRank();
-    public final class TaxonomicRank extends Property<TaxonomicRanks> implements FromAny, ToOne {
+    public final Rank rank = new Rank();
+    public final class Rank extends Property<TaxonomicRank> implements FromAny, ToAtMostOne {
 
-      private TaxonomicRank() { super(TaxonomicRanks.class); }
+      private Rank() { super(TaxonomicRank.class); }
     }
   }
 
-  /*
-    The set of valid ranks is nowhere documented; this is just an approximation.
-  */
-  public static enum TaxonomicRanks {
-
-    noRank,
-    superkingdom,
-    kingdom,
-    superphylum,
-    phylum,
-    subphylum,
-    clazz, // reserved word
-    subclass,
-    superclass,
-    infraclass,
-    order,
-    parvorder,
-    suborder,
-    infraorder,
-    family,
-    subfamily,
-    superfamily,
-    tribe,
-    subtribe,
-    genus,
-    subgenus,
-    speciesGroup,
-    speciesSubgroup,
-    species,
-    subspecies,
-    varietas,
-    forma,
-    thereIsAnIndeterminateNumberOfRanksAsATaxonomistMayInventANewRankAtWillAtAnyTimeIfTheyFeelThisIsNecessary;
+  /* This is the set of ranks stored in Bio4j. There is no general list of valid ranks, because there is an indeterminate number of ranks as a taxonomist may invent a new rank at will at any time if they feel this is necessary. */
+  public static enum TaxonomicRank {
```

Comment:
Why? `rank` property maps `Taxon` to `TaxonomicRank`. If it was in plural, it would be confusing.